xwayland/xwm.c

```diff
@@ -1164,6 +1176,12 @@ static void xwm_handle_net_wm_state_message(struct wlr_xwm *xwm,
 
 		wlr_signal_emit_safe(&xsurface->events.request_maximize, xsurface);
 	}
+
+	if (minimized != xsurface->minimized) {
+		xsurface->saved_width = xsurface->width;
```

Comment:
We should only do this if the surface is minimized.

Reply:
Please be more specific in your comments.
Do what emit signals? Safe state?
The handling is exactly the same as for maximized in this function. 

I suppose the below is what you think is right so I changed it to that:
`	if (minimized && minimized != xsurface->minimized) {
`

Reply:
No, we want to emit the signal even if the client requests to *not* be minimized.

Please have a look at the similar block of code above for maximized. We only set `saved_width` and `saved_height` if the surface becomes maximized. We should do the same here, only update the saved size if the surface wants to become minimized.

Reply:
Can you look into this?

Reply:
I don't see how this is wrong, maybe you suggest a proper way?